src/apps/competitions/tasks.py

```diff
@@ -744,3 +745,15 @@ def update_phase_statuses():
     competitions = Competition.objects.exclude(phases__in=Phase.objects.filter(is_final_phase=True, end__lt=now()))
     for comp in competitions:
         comp.update_phase_statuses()
+
+
+@app.task(queue='site-worker')
+def submission_status_cleanup():
+    submissions = Submission.objects.filter(status=Submission.RUNNING).select_related('phase')
+
+    for sub in submissions:
+        # Check if the submission has been running for 24 hours longer than execution_time_limit
+        if sub.created_when < now() - timedelta(milliseconds=(3600000 * 24) + sub.phase.execution_time_limit):
```

Comment:
hmm, I wonder if `created_when` works properly here, what if a submission was created, queued up for a day or two, then started? do we need a `started_when`?

Reply:
Yeah that could definitely happen. Adding a `started_when`